kafka-handler/src/java/org/apache/hadoop/hive/kafka/KafkaUtils.java

```diff
@@ -397,4 +421,31 @@ static void addKerberosJaasConf(Configuration configuration, Properties props) {
     }
     log.info("Kafka client running with following JAAS = [{}]", jaasConf);
   }
+
+  /**
+   * Returns the security protocol if one is defined in the properties and null otherwise.
+   * <p>The following properties are examined to determine the protocol:</p>
+   * <ol>
+   *   <li>security.protocol</li>
+   *   <li>kafka.consumer.security.protocol</li>
+   *   <li>kafka.producer.security.protocol</li>
+   * </ol>
+   * <p>and the first non null/not empty is returned.</p>
+   * <p>Defining multiple security protocols at the same time is invalid but this method is lenient and tries to pick
+   * the most reasonable option.</p>
+   * @param props the properties from which to obtain the protocol.
+   * @return the security protocol if one is defined in the properties and null otherwise.
+   */
+  static SecurityProtocol securityProtocol(Properties props) {
+    List<String> securityProtocolConfigs = Arrays.asList(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG,
+        CONSUMER_CONFIGURATION_PREFIX + "." + CommonClientConfigs.SECURITY_PROTOCOL_CONFIG,
+        PRODUCER_CONFIGURATION_PREFIX + "." + CommonClientConfigs.SECURITY_PROTOCOL_CONFIG);
```

Comment:
Why you need ``Arrays.asList``, can't we directly have an array?
```
    String[] securityProtocolConfigs = new String[] {CommonClientConfigs.SECURITY_PROTOCOL_CONFIG,
        CONSUMER_CONFIGURATION_PREFIX + "." + CommonClientConfigs.SECURITY_PROTOCOL_CONFIG,
        PRODUCER_CONFIGURATION_PREFIX + "." + CommonClientConfigs.SECURITY_PROTOCOL_CONFIG};
```

Reply:
Fixed in https://github.com/apache/hive/pull/4430/commits/fcc14e85f07b69b43154911fe2ea766fdbc5c3fb